python-project/mortar/image/threshold.py

```diff
@@ -16,25 +15,17 @@
 class Threshold(Filter):
     """
     Perform thresholding on an image.
-
-    TODO this class doesn't yet support the necessary thresholding options.
     """
     name = 'Threshold'
 
-    def run(self, input: PILImage) -> Optional[PILImage]:
+    def run(self,
+            input: PILImage,
```

Comment:
I was testing the updated filter in the pipeline and noticed that it doesn't seem to work correctly with RGB input images. It does process and return a new image, but the result isn't as expected, so I think we need to treat input images in non-RGB colorspace as an error condition.

Can you add an input check that makes this method return `None` if the image isn't grayscale? See the `Image.mode` attribute in the pillow library docs.